packages/block-editor/src/hooks/metadata-name.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+/**
+ * Internal dependencies
+ */
+import { hasBlockMetadataSupport } from './metadata';
+
+/**
+ * Filters registered block settings, adding an `__experimentalLabel` callback if one does not already exist.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addLabelCallback( settings ) {
+	// If blocks provide their own label callback, do not override it.
+	if ( settings.__experimentalLabel ) {
+		return settings;
+	}
+
+	const supportsBlockNaming = hasBlockMetadataSupport(
+		settings,
+		'name',
+		false
```

Comment:
a comment about what `false` is would be nice.

Reply:
```suggestion
		false // default value
```